### ISSUE TITLE
Name price API call makes wrong estimate

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#632]
Testing name registration using 0.17.0.0 on mainnet, I noticed that the price api endpoint assumes we are sending 4 transactions:

* preorder
* register
* update
* transfer

It also assume we're using a multi-sig address.

As a result, the estimate was much higher than the actual cost for registration via the browser.

We don't use currently multi-sig addresses in the browser and the `POST /v1/names` API as of Hard Fork Day 2017 conducts a 2 transaction registration.

```
127.0.0.1 - - [06/Oct/2017 18:47:50] "GET /v1/names/happyforkday.id HTTP/1.1" 404 -
[2017-10-06 18:47:50,410] [DEBUG] [rpc:3917] (12010.123145320943616) 
full path: /v1/prices/names/happyforkday.id
method: GET
path: /v1/prices/names/happyforkday.id
qs: {}
headers:
origin: http://localhost:3000
accept-language: en-US,en;q=0.8
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36
host: localhost:6270
referer: http://localhost:3000/profiles/i/add-username/18EJpP1T42rbsCwJPSj9XUUUzZa9Sm9uF3/search

[2017-10-06 18:47:50,522] [DEBUG] [rpc:390] (12010.123145320943616) No authorization header
[2017-10-06 18:47:50,634] [DEBUG] [rpc:3959] (12010.123145320943616) No authentication needed
[2017-10-06 18:47:50,752] [DEBUG] [proxy:222] (12010.123145320943616) Default proxy to https://node.blockstack.org:6263
[2017-10-06 18:47:50,862] [DEBUG] [client:95] (12010.123145320943616) Connect to https://node.blockstack.org:6263
[2017-10-06 18:47:50,975] [WARNING] [dropbox:188] (12010) Config file '/Users/larry/Library/Application Support/Blockstack/config/client.ini': section 'dropbox' is missing 'token'.  Write access will be disabled
[2017-10-06 18:47:50,976] [DEBUG] [client:110] (12010.123145320943616) Loaded storage drivers ['disk', 'dropbox', 's3', 'blockstack_resolver', 'blockstack_server', 'http', 'dht']
[2017-10-06 18:47:51,088] [DEBUG] [rpc:5501] (12010.123145320943616) RPC running (/Users/larry/Library/Application Support/Blockstack/config/api_endpoint.pid)
[2017-10-06 18:47:51,199] [DEBUG] [actions:715] (12010.123145320943616) Try to get wallet keys from API server
[2017-10-06 18:47:51,315] [DEBUG] [proxy:222] (12010.123145320943616) Default proxy to https://node.blockstack.org:6263
[2017-10-06 18:47:51,428] [DEBUG] [client:95] (12010.123145320943616) Connect to https://node.blockstack.org:6263
[2017-10-06 18:47:51,544] [WARNING] [dropbox:188] (12010) Config file '/Users/larry/Library/Application Support/Blockstack/config/client.ini': section 'dropbox' is missing 'token'.  Write access will be disabled
[2017-10-06 18:47:51,549] [DEBUG] [client:110] (12010.123145320943616) Loaded storage drivers ['disk', 'dropbox', 's3', 'blockstack_resolver', 'blockstack_server', 'http', 'dht']
[2017-10-06 18:47:51,660] [DEBUG] [session:189] (12002.123145320943616) Connect to bitcoind at bitcoin.blockstack.com:8332 (/Users/larry/Library/Application Support/Blockstack/config/client.ini)
[2017-10-06 18:47:51,660] [DEBUG] [session:104] (12002.123145320943616) [12010] Connect to bitcoind at http://blockstack@bitcoin.blockstack.com:8332, timeout=300.0
[2017-10-06 18:47:52,334] [DEBUG] [fees:87] (12002.123145320943616) Bitcoin estimatefee(2) is 0.00172686
[2017-10-06 18:47:52,335] [DEBUG] [fees:93] (12002.123145320943616) Bitcoin estimatefee(2) is 0.00172686 (169 satoshi/byte)
[2017-10-06 18:47:52,336] [DEBUG] [actions:620] (12010.123145320943616) Get operation fees for preorder, register, update
[2017-10-06 18:47:52,471] [DEBUG] [safety:755] (12002.123145320943616) Getting UTXOs for 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp
[2017-10-06 18:47:52,471] [DEBUG] [insight_api:41] (12002.123145320943616) GET https://utxo.blockstack.org/insight-api/addr/33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp/utxo
[2017-10-06 18:47:54,142] [DEBUG] [insight_api:52] (12002.123145320943616) 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp has 1 UTXOs
[2017-10-06 18:47:54,143] [DEBUG] [insight_api:41] (12002.123145320943616) GET https://utxo.blockstack.org/insight-api/addr/3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP/utxo
[2017-10-06 18:47:55,057] [DEBUG] [insight_api:52] (12002.123145320943616) 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP has 2 UTXOs
[2017-10-06 18:47:55,061] [DEBUG] [safety:767] (12002.123145320943616) Balance of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP is 58724 satoshis
[2017-10-06 18:47:55,062] [DEBUG] [safety:788] (12002.123145320943616) Get total operation fees for running 'preorder,register,update' on happyforkday.id owned by 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp paid by 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,062] [DEBUG] [safety:796] (12002.123145320943616) Operation preorder may consume up to 1 owner inputs of 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp and 2 payment inputs of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,063] [DEBUG] [safety:796] (12002.123145320943616) Operation register may consume up to 1 owner inputs of 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp and 1 payment inputs of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,063] [DEBUG] [safety:796] (12002.123145320943616) Operation update may consume up to 1 owner inputs of 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp and 1 payment inputs of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,063] [DEBUG] [safety:1228] (12002.123145320943616) Add task preorder_tx_fee (<function _estimate_preorder_tx at 0x109b3f500>)
[2017-10-06 18:47:55,064] [DEBUG] [safety:1228] (12002.123145320943616) Add task register_tx_fee (<function _estimate_register_tx at 0x109b3f578>)
[2017-10-06 18:47:55,064] [DEBUG] [safety:1228] (12002.123145320943616) Add task update_tx_fee (<function _estimate_update_tx at 0x109b3f410>)
[2017-10-06 18:47:55,064] [DEBUG] [utils:337] (12010.123145320943616) Start task 'update_tx_fee'
[2017-10-06 18:47:55,173] [DEBUG] [utils:283] (12010.123145325150208) Run task <functools.partial object at 0x109a2baa0>
[2017-10-06 18:47:55,173] [DEBUG] [utils:337] (12010.123145320943616) Start task 'register_tx_fee'
[2017-10-06 18:47:55,294] [DEBUG] [nameops:175] (12010.123145325150208) Try building a NAME_UPDATE with inputs 0-1 of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,406] [DEBUG] [utils:283] (12010.123145329356800) Run task <functools.partial object at 0x109a2bb50>
[2017-10-06 18:47:55,407] [DEBUG] [utils:337] (12010.123145320943616) Start task 'preorder_tx_fee'
[2017-10-06 18:47:55,522] [ERROR] [nameops:184] (12010.123145325150208) Not enough inputs for transaction (total: 22000, to spend: 5500, fee: 21000).
Traceback (most recent call last):
  File "/tmp/blockstack-venv/lib/python2.7/site-packages/blockstack_client/backend/nameops.py", line 177, in make_cheapest_nameop
    unsigned_tx = tx_builder(*tx_args, **tx_kw)
  File "/tmp/blockstack-venv/lib/python2.7/site-packages/blockstack_client/tx.py", line 94, in update_tx
    inputs, outputs = tx_update(*args, **kw)
  File "/tmp/blockstack-venv/lib/python2.7/site-packages/blockstack_client/operations/update.py", line 147, in make_transaction
    outputs = make_outputs( nulldata, inputs, owner_addr, tx_fee, pay_fee=pay_fee )
  File "/tmp/blockstack-venv/lib/python2.7/site-packages/blockstack_client/operations/update.py", line 118, in make_outputs
    "value": virtualchain.calculate_change_amount(inputs, op_fee, dust_fee)}
  File "/tmp/blockstack-venv/lib/python2.7/site-packages/virtualchain/lib/blockchain/bitcoin_blockchain/keys.py", line 355, in calculate_change_amount
    raise ValueError('Not enough inputs for transaction (total: {}, to spend: {}, fee: {}).'.format(total_amount_in, send_amount, fee))
ValueError: Not enough inputs for transaction (total: 22000, to spend: 5500, fee: 21000).
[2017-10-06 18:47:55,655] [DEBUG] [nameops:175] (12010.123145329356800) Try building a NAME_REGISTRATION with inputs 0-1 of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:55,759] [DEBUG] [utils:283] (12010.123145333563392) Run task <functools.partial object at 0x109a2bba8>
[2017-10-06 18:47:55,759] [DEBUG] [utils:337] (12010.123145320943616) Start task 'get_balance'
[2017-10-06 18:47:55,763] [WARNING] [nameops:606] (12010.123145325150208) No UTXOs for 33GcdiK7JxuNX8NCQTvVh58dZEKW2ym7Hp and/or 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP; estimating...
[2017-10-06 18:47:55,879] [DEBUG] [nameops:180] (12010.123145329356800) Funded NAME_REGISTRATION with inputs 0-1 of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:56,108] [DEBUG] [utils:283] (12010.123145337769984) Run task <function _get_balance at 0x107b51320>
[2017-10-06 18:47:56,109] [DEBUG] [utils:344] (12010.123145320943616) Join task 'get_balance'
[2017-10-06 18:47:56,365] [DEBUG] [fees:142] (12002.123145325150208) Multisig private key makes ~355 byte signature
[2017-10-06 18:47:56,478] [DEBUG] [fees:142] (12002.123145325150208) Multisig private key makes ~355 byte signature
[2017-10-06 18:47:56,478] [DEBUG] [nameops:635] (12010.123145325150208) update tx 935 bytes, 158015 satoshis txfee
[2017-10-06 18:47:56,538] [DEBUG] [nameops:449] (12010.123145329356800) Unsigned tx: 
[2017-10-06 18:47:56,654] [DEBUG] [utils:285] (12010.123145337769984) Task exit <function _get_balance at 0x107b51320>
[2017-10-06 18:47:56,887] [DEBUG] [nameops:636] (12010.123145325150208) signed subsidized tx: 
[2017-10-06 18:47:57,155] [DEBUG] [fees:142] (12002.123145329356800) Multisig private key makes ~355 byte signature
[2017-10-06 18:47:57,156] [DEBUG] [nameops:479] (12010.123145329356800) register tx 499 bytes, 84331 satoshis txfee
[2017-10-06 18:47:57,173] [DEBUG] [nameops:175] (12010.123145333563392) Try building a NAME_PREORDER with inputs 0-1 of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:57,268] [DEBUG] [utils:344] (12010.123145320943616) Join task 'register_tx_fee'
[2017-10-06 18:47:57,386] [DEBUG] [nameops:641] (12010.123145325150208) Additional update dust fee: 21000
[2017-10-06 18:47:57,500] [DEBUG] [nameops:480] (12010.123145329356800) signed subsidized tx: 
[2017-10-06 18:47:57,613] [DEBUG] [nameops:180] (12010.123145333563392) Funded NAME_PREORDER with inputs 0-1 of 3Ht2PzmsT4gy4EpBTp7FKco239ZiHvXdUP
[2017-10-06 18:47:57,836] [DEBUG] [utils:285] (12010.123145325150208) Task exit <functools.partial object at 0x109a2baa0>
[2017-10-06 18:47:57,954] [WARNING] [bits:317] (12002.123145329356800) Did not parse entire tx (355 bytes remaining)
[2017-10-06 18:47:57,955] [DEBUG] [nameops:485] (12010.123145329356800) Additional register dust fee: 26500
[2017-10-06 18:47:58,213] [DEBUG] [fees:142] (12002.123145333563392) Multisig private key makes ~355 byte signature
[2017-10-06 18:47:58,214] [DEBUG] [nameops:408] (12010.123145333563392) preorder tx 522 bytes, 88218 satoshis
[2017-10-06 18:47:58,409] [DEBUG] [utils:285] (12010.123145329356800) Task exit <functools.partial object at 0x109a2bb50>
[2017-10-06 18:47:58,524] [DEBUG] [nameops:409] (12010.123145333563392) signed subsidized tx: 
[2017-10-06 18:47:58,639] [DEBUG] [utils:344] (12010.123145320943616) Join task 'preorder_tx_fee'
[2017-10-06 18:47:58,754] [WARNING] [bits:317] (12002.123145333563392) Did not parse entire tx (355 bytes remaining)
[2017-10-06 18:47:58,756] [DEBUG] [nameops:414] (12010.123145333563392) Additional preorder dust fee: 26500
[2017-10-06 18:47:58,984] [DEBUG] [utils:285] (12010.123145333563392) Task exit <functools.partial object at 0x109a2bba8>
[2017-10-06 18:47:59,098] [DEBUG] [utils:344] (12010.123145320943616) Join task 'update_tx_fee'
[2017-10-06 18:47:59,209] [DEBUG] [safety:1278] (12002.123145320943616) Balance is 58724 satoshis
[2017-10-06 18:47:59,210] [DEBUG] [safety:1306] (12002.123145320943616) preorder_tx_fee +114718 satoshis (tx fee)
[2017-10-06 18:47:59,210] [DEBUG] [safety:1310] (12002.123145320943616) preorder_tx_fee +2500 satoshis (name cost)
[2017-10-06 18:47:59,211] [DEBUG] [safety:1306] (12002.123145320943616) register_tx_fee +110831 satoshis (tx fee)
[2017-10-06 18:47:59,211] [DEBUG] [safety:1306] (12002.123145320943616) update_tx_fee +179015 satoshis (tx fee)
[2017-10-06 18:47:59,212] [DEBUG] [safety:1320] (12002.123145320943616) Total cost of preorder,register,update is 407064 satoshis
[2017-10-06 18:47:59,212] [DEBUG] [safety:1325] (12002.123145320943616) Insufficient funds: need 407064, have 58724
127.0.0.1 - - [06/Oct/2017 18:47:59] "GET /v1/prices/names/happyforkday.id HTTP/1.1" 200 -
```

**Comment by kantai** [comment]
With regards to switching from estimating multi-sig to single-sig -- 

Currently, the price lookup endpoint tries to figure out how many UTXOs will fund the transactions (via the core wallet address) -- obviously, right now, that estimation is wrong because the funding addresses will be different. So instead, I'll use a dummy single-sing address / pk rather than forcing the client to pass that information --- to get single_sig, pass the query string `single_sig=1`

```
 $ curl -H "Origin: http://localhost:3000" -H "Authorization: bearer password" -H "Content-Type: application/json" http://localhost:6270/v1/prices/names/aaron.id?single_sig=1 -s | python -m json.tool
{
    "name_price": {
        "btc": 0.0004,
        "satoshis": 40000
    },
    "preorder_tx_fee": {
        "btc": 0.00073313,
        "satoshis": 73313
    },
    "register_tx_fee": {
        "btc": 0.00068581,
        "satoshis": 68581
    },
    "total_estimated_cost": {
        "btc": 0.00181894,
        "satoshis": 181894
    },
    "total_tx_fees": 141894,
    "warnings": [
        "Insufficient funds; fees are rough estimates."
    ]
}
```

This is in the hotfix branch, I'll test it and then merge to master

**Comment by larrysalibra** [comment]
Thanks @kantai! I made the change to the browser: https://github.com/blockstack/blockstack-browser/commit/8aab06e8c9ea1dd6a6d5a079fc22f1e5372fcb3e

**Comment by kantai** [comment]
Great! fixed in blockstack-core v0.17.0.2
